core/src/main/scala/org/apache/spark/sql/delta/actions/DeletionVectorDescriptor.scala

```diff
@@ -0,0 +1,254 @@
+  }
+
+  /** Returns the estimated number of bytes required to serialize this object. */
+  @JsonIgnore
+  protected[delta] lazy val estimatedSerializedSize: Int = {
+    // (cardinality(8) + sizeInBytes(4)) + storageType + pathOrInlineDv + option[offset(4)]
+    12 + storageType.length + pathOrInlineDv.length + (if (offset.isDefined) 4 else 0)
+  }
+}
+
+object DeletionVectorDescriptor {
+
+  // TODO: This is a temporary solution that allows us to add new protocol versions for other
+  // things, while DVs are still in development.
+  final val REQUIRED_READER_VERSION = Int.MaxValue
+  final val REQUIRED_WRITER_VERSION = Int.MaxValue
+
+  /** String that is used in all file names generated by deletion vector store */
+  val DELETION_VECTOR_FILE_NAME_CORE = "deletion_vector"
+
+  // Markers to separate different kinds of DV storage.
+  final val PATH_DV_MARKER: String = "p"
+  final val INLINE_DV_MARKER: String = "i"
+  final val UUID_DV_MARKER: String = "u"
+
+  final val STRUCT_TYPE: StructType = new StructType()
+    .add("storageType", StringType)
+    .add("pathOrInlineDv", StringType)
+    .add("offset", IntegerType)
+    .add("sizeInBytes", IntegerType, nullable = false)
+    .add("cardinality", LongType, nullable = false)
+
+  private lazy val _encoder = new DeltaEncoder[DeletionVectorDescriptor]
+  implicit def encoder: Encoder[DeletionVectorDescriptor] = _encoder.get
+
+  /** Utility method to create an on-disk [[DeletionVectorDescriptor]] */
+  def onDiskWithRelativePath(
+      id: UUID,
+      randomPrefix: String = "",
+      sizeInBytes: Int,
+      cardinality: Long,
+      offset: Option[Int] = None): DeletionVectorDescriptor =
+    DeletionVectorDescriptor(
+      storageType = UUID_DV_MARKER,
+      pathOrInlineDv = encodeUUID(id, randomPrefix),
+      offset = offset,
+      sizeInBytes = sizeInBytes,
+      cardinality = cardinality)
+
+  /** Utility method to create an on-disk [[DeletionVectorDescriptor]] */
+  def onDiskWithAbsolutePath(
+      path: String,
+      sizeInBytes: Int,
+      cardinality: Long,
+      offset: Option[Int] = None): DeletionVectorDescriptor =
+    DeletionVectorDescriptor(
+      storageType = PATH_DV_MARKER,
+      pathOrInlineDv = path,
+      offset = offset,
+      sizeInBytes = sizeInBytes,
+      cardinality = cardinality)
+
+  /** Utility method to create an inline [[DeletionVectorDescriptor]] */
+  def inlineInLog(
+      data: Array[Byte],
+      cardinality: Long): DeletionVectorDescriptor =
+    DeletionVectorDescriptor(
+      storageType = INLINE_DV_MARKER,
+      pathOrInlineDv = encodeData(data),
+      sizeInBytes = data.length,
+      cardinality = cardinality)
+
+  /**
+   * Return the unique path under `parentPath` that is based on `id`.
+   *
+   * Optionally, prepend a `prefix` to the name.
+   */
+  def assembleDeletionVectorPath(targetParentPath: Path, id: UUID, prefix: String = ""): Path = {
+    val core = DELETION_VECTOR_FILE_NAME_CORE
```

Comment:
Is it really worthwhile to assign a field to a (shorter) field? Why not just use the static field directly?